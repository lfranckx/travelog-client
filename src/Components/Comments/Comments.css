.comments-section {
    margin-top: 10px;
}

ul > li {
    list-style-type: none;
}

.comments-list {
    padding: 0;
}

.comment {
    display: flex;
    align-items: top;
    flex-direction: row;
    width: 95%;
    padding: 10px;
    font-size: 1rem;
    border-radius: 3px;
    margin-bottom: 10px;
}

.comment > a {
    display: flex;
    align-items: top;
    margin-right: 10px;
    position: relative;
    top: 12px;
}

.comment-profile {
    height: 34px;
    width: 34px;
    border-radius: 50%;
    margin-right: 5px;
}

.comment-user {
    margin: 0;
    position: relative;
    top: 5px;
}

.comment-text {
    margin: 0;
    background-color: #efefef;
    padding: 15px;
    white-space: pre-line;
}

.comment-form textarea {
    padding: 10px;
}

.post {
    padding: 6px 16px;
}
