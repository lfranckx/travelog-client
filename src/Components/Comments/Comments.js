import React, { Component } from 'react';
import './Comments.css';
import ArticleContext from '../../Contexts/ArticleContext';
import ArticleApiService from '../../Services/article-api-service';
import trash from '../../icons/trash.png';

export default class Comments extends Component {
    static contextType = ArticleContext;

    handleSubmit = ev => {
        ev.preventDefault();
        const { article } = this.props;
        const { user } = this.context;
        const { comment } = ev.target;
        const newComment = {
            comment: comment.value,
            username: user.username,
            author_name: user.name,
            article_id: article.id
        };

        ArticleApiService.postComment(newComment)
            .then(this.context.addComment)
            .then(() => {
                comment.value = ''
            })
            .catch(this.context.setError);
    } 

    render() {
        const { comments } = this.props;
        return (
            <section className="comments-section">
                <form className="comment-form"
                    onSubmit={this.handleSubmit}>
                    <div>
                        <textarea rows="5"
                            name="comment"
                            aria-label="comment"
                            className="comment"
                            placeholder="Leave a comment"
                            required
                        >
                        </textarea>
                    </div>
                    <button type="submit">Post</button>
                </form>
                <ArticleComments comments={comments} /> 
            </section>
        )
    }
}

function ArticleComments({ comments = []}) {
    return (
        <ul className="comments-list">
            {comments.map(comment =>
                <li key={comment.id} className="comment">
                    <p className="comment-text">{comment.comment}</p>
                    <p className="comment-user">{comment.username}</p>
                </li>
            )}
        </ul>
    );
}
