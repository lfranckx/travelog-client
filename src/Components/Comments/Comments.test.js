import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Comments from './Comments';

describe(`Comments`, () => {
    const props = {
        article_id: 1,
        author_name: "Sam Smith",
        comment: "Hey dude!",
        date: "2020-05-18T08:32:54.586Z",
        id: 1,
        profile_image: 'http://placehold.it/500x500',
    };

    it(`renders an article given props`, () => {
        const wrapper = shallow(<Comments {...props}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});