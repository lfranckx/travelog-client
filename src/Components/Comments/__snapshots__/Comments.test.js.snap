// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments renders an article given props 1`] = `<ArticleComments />`;
